processing/src/main/java/io/druid/segment/IndexMerger.java

```diff
@@ -470,20 +480,71 @@ public String apply(@Nullable String input)
       }
     };
 
-    return makeIndexFiles(indexes, outDir, progress, mergedDimensions, mergedMetrics, null, rowMergerFn, indexSpec);
+    return makeIndexFiles(
+        indexes,
+        mergeAggregators(indexes),
+        outDir,
+        progress,
+        mergedDimensions,
+        mergedMetrics,
+        null,
+        rowMergerFn,
+        indexSpec
+    );
+  }
+
+  public static final AggregatorFactory[] mergeAggregators(List<IndexableAdapter> indexes)
```

Comment:
sure, will update the javadoc.

I have tried to replicate the behavior at of merging name of metrics at https://github.com/druid-io/druid/blob/master/processing/src/main/java/io/druid/segment/IndexMerger.java#L435 .
If two different aggregators had same name, then the first one will be taken. Also, if any of the input index did not have aggregators in the metadata then resulting segment will not have same as well.
does this behavior look ok or we should do something else?

Also, I'm yet to write the UTs for this. sent the PR first to ensure that the behavior chosen looks ok to others.